pkg/controllers/kube-controller-manager.go

```diff
@@ -16,31 +16,59 @@ limitations under the License.
 package controllers
 
 import (
+	"context"
 	"path/filepath"
 	"strconv"
 
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
 
 	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util"
 
+	"k8s.io/component-base/cli/globalflag"
 	kubecm "k8s.io/kubernetes/cmd/kube-controller-manager/app"
+	kubecmoptions "k8s.io/kubernetes/cmd/kube-controller-manager/app/options"
 )
 
-func KubeControllerManager(cfg *config.MicroshiftConfig) {
-	command := kubecm.NewControllerManagerCommand()
+type KubeControllerManager struct {
+	kubecmOptions *kubecmoptions.KubeControllerManagerOptions
+	kubeconfig    string
+}
+
+func NewKubeControllerManager(cfg *config.MicroshiftConfig) *KubeControllerManager {
+	s := &KubeControllerManager{}
+	s.configure(cfg)
+	return s
+}
+
+func (s *KubeControllerManager) Name() string           { return "kube-controller-manager" }
+func (s *KubeControllerManager) Dependencies() []string { return []string{"kube-apiserver"} }
+
+func (s *KubeControllerManager) configure(cfg *config.MicroshiftConfig) {
+	caCertFile := filepath.Join(cfg.DataDir, "certs", "ca-bundle", "ca-bundle.crt")
+	kubeconfig := filepath.Join(cfg.DataDir, "resources", "kubeadmin", "kubeconfig")
```

Comment:
can you use kube-controller-manager kubeconfig?

Reply:
yes, I wasn't sure which to use. I'll update it.

Reply:
updated  to _dataDir_/resources/kube-controller-manager/kubeconfig